sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -930,17 +937,25 @@ private void ValidateCanCreateSenderLink(string entityPath)
 
             closeHandler = (snd, args) =>
             {
-                ActiveLinks.TryRemove(link, out var timer);
-
-                timer?.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
-                timer?.Dispose();
+                EndTrackingLinkAsActive(link);
 
                 link.Closed -= closeHandler;
             };
 
             link.Closed += closeHandler;
         }
 
+        private void EndTrackingLinkAsActive(AmqpObject link)
```

Comment:
Would `StopTrackingLinkAsActive` be a more descriptive name?

Reply:
I mostly used this for the Begin/End consistency, but agreed that it reads strangely. I will see about renaming both methods.